### PR TITLE
[ao][sparsity] Implemented state dict and serialization functionalities

### PR DESCRIPTION
**Pull request by macandro96**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79535
* #79534
* #79254
* #79253
* __->__ #79427
* #79252
* #79251

The state of the data sparsifier object contains the name->mask mapping, name -> config mapping and the state_dict() of the container.
The load_state_dict() and __set_state__() automatically creates a container moduie and loads the named data internally without having the user to intervene.

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

Differential Revision: [D37164117](https://our.internmc.facebook.com/intern/diff/D37164117)

### CODE REVIEW

**Review by z-a-f — APPROVED** [review]
lgtm

**Comment by macandro96** [comment]
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37164117).

**Comment by macandro96** [comment]
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37164117).

**Comment by macandro96** [comment]
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37164117).
